src/utils/__tests__/actions.spec.js

```diff
@@ -306,34 +331,108 @@ describe('actions', () => {
+    it('should call executeAutomation when automationTriggered is true', async () => {
       inputs = [];
+      const appId = 'fakeAppId';
+      let automation;
       automationPostData = false;
+      automationTriggered = true;
+      buttonId = 'fakeButtonId';
+      const time = new Date(2022, 10, 1);
+      jest.useFakeTimers('modern');
+      jest.setSystemTime(time);
       const actionObject = actions.find((action) => action.value === 'executeAutomation');
-      await actionObject.getActionCall({ app, automation, automationPostData })();
-      expect(global.fetch).toHaveBeenCalledThrice;
-      expect(global.fetch).toHaveBeenCalledWith(`../api/v1/items/${automation}`);
-      expect(global.fetch).toHaveBeenCalledWith(`../api/v1/automations/${resourceId}`);
-      expect(global.fetch).toHaveBeenCalledWith(
-        `../api/v1/automations/${id}/actions/execute?X-Execution-Token=${executionToken}`
-      );
+      await actionObject.getActionCall({ app, automation, automationId, automationTriggered, automationExecutionToken, automationPostData, buttonId })();
+      expect(global.fetch).toHaveBeenCalledTimes(4);
+      expect(global.fetch).toHaveBeenCalledWith(`../api/v1/users/me`);
+      expect(global.fetch).toHaveBeenCalledWith(`../api/v1/apps/${appId}`);
+      expect(global.fetch).toHaveBeenCalledWith('../api/v1/csrf-token');
+      const headers = {
+        'Content-Type': 'application/json',
+        'qlik-csrf-token': csrfToken,
+        'X-Execution-Token': automationExecutionToken
+      }
+      const postOptions = {
+        method: 'POST',
+        headers,
+        body: JSON.stringify({
+          app: appId,
+          bookmark: '',
+          sheet: sheetId,
+          user: subject,
+          space: spaceId,
+          tenant: tenantId,
+          time
+        }),
+      }
+      expect(global.fetch).toHaveBeenCalledWith(`../api/v1/automations/${automationId}/actions/execute?X-Execution-Token=${automationExecutionToken}`, postOptions);
+      jest.useRealTimers();
+    });
+
+    it('should call automation run when automationTriggered is false', async () => {
+      inputs = [];
+      let automation;
+      automationPostData = false;
+      automationTriggered = false;
+      buttonId = 'fakeButtonId';
+      const time = new Date(2022, 10, 1);
+      jest.useFakeTimers('modern');
+      jest.setSystemTime(time);
+      const actionObject = actions.find((action) => action.value === 'executeAutomation');
+      await actionObject.getActionCall({ app, automation, automationId, automationTriggered, automationExecutionToken, automationPostData, buttonId })();
+      expect(global.fetch).toHaveBeenCalledTimes(4);
+      expect(global.fetch).toHaveBeenCalledWith(`../api/v1/users/me`);
+      expect(global.fetch).toHaveBeenCalledWith(`../api/v1/apps/${app.id}`);
+      expect(global.fetch).toHaveBeenCalledWith('../api/v1/csrf-token');
+      const automationData = {
+        guid: automationId,
```

Comment:
**suggestion:** we can use `id: automationId` I think. The API still accepts `guid`, but it's an old property and `id` should be used if possible.